addons/survey/static/src/views/survey_views.js

```diff
@@ -0,0 +1,82 @@
+/** @odoo-module **/
+
+import { registry } from '@web/core/registry';
+import { ListRenderer } from "@web/views/list/list_renderer";
+import { KanbanRenderer } from "@web/views/kanban/kanban_renderer";
+import { listView } from '@web/views/list/list_view';
+import { kanbanView } from '@web/views/kanban/kanban_view';
+import { useService } from "@web/core/utils/hooks";
+
+const { useComponent, useEffect, useRef } = owl;
+
+export function useSurveyLoadSampleHook(selector) {
+    const component = useComponent();
+    component.rootRef = useRef("root");
+    component.actionService = useService("action");
+    component.orm = useService('orm');
+    component.isLoadingSample = false;
+    useEffect(
```

Comment:
I don't know owl so this part confuses me greatly ^^

Not sure what useEffect does exactly, does it take a function as 1st argument and the arguments to pass to it as 2nd argument? It looks very strange.
And the part where you return another function that removes the eventHandler... -> 🤯 

Probably a stupid suggestion but how about a simple "onMounted", that way we don't have to worry about removing the handlers?

```js
onMounted(
    () => {
        const sampleTypes = component.rootRef.el ? component.rootRef.el.querySelectorAll(selector) : [];
        for (const sampleType of sampleTypes) {
            // (would let 'loadSample' extract the method to make this more simple)
            sampleType.addEventListener('click', component.loadSample);
        }
    }
)
```

Reply:
We could have used onMounted, but we would have had to add onPatched aswell doing the same thing.
Also doing `component.loadSample` means we have to implement the function on the component each time we want to use the hook, which isn't great IMO


`useEffect` takes two functions, the first one is the effect which should return the cleanup, the second one is the dependencies.
`useEffect` is really well explained [here](https://github.com/odoo/owl/blob/master/doc/reference/hooks.md#useeffect)